pkg/kubelet/cloudresource/cloud_request_manager.go

```diff
@@ -57,68 +55,69 @@ type cloudResourceSyncManager struct {
 // from a cloud provider through requests that are sensitive to timeouts and hanging
 func NewSyncManager(cloud cloudprovider.Interface, nodeName types.NodeName, syncPeriod time.Duration) SyncManager {
 	return &cloudResourceSyncManager{
-		cloud:      cloud,
-		syncPeriod: syncPeriod,
-		nodeName:   nodeName,
+		cloud:                cloud,
+		syncPeriod:           syncPeriod,
+		nodeName:             nodeName,
+		nodeAddressesMonitor: sync.NewCond(&sync.Mutex{}),
 	}
 }
 
-func (manager *cloudResourceSyncManager) getNodeAddressSafe() ([]v1.NodeAddress, error) {
-	manager.nodeAddressesMux.Lock()
-	defer manager.nodeAddressesMux.Unlock()
-
-	return manager.nodeAddresses, manager.nodeAddressesErr
-}
-
-func (manager *cloudResourceSyncManager) setNodeAddressSafe(nodeAddresses []v1.NodeAddress, err error) {
-	manager.nodeAddressesMux.Lock()
-	defer manager.nodeAddressesMux.Unlock()
-
-	manager.nodeAddresses = nodeAddresses
-	manager.nodeAddressesErr = err
-}
-
 // NodeAddresses does not wait for cloud provider to return a node addresses.
```

Comment:
Update the comment to reflect new behavior. This method could block for a while

Reply:
Done.